src/arm/soc.c

```diff
@@ -676,6 +677,11 @@ bool match_special(char* soc_name, struct system_on_chip* soc) {
     return true;
   }
 
+  if(strcmp(soc_name, "Snapdragon (TM) 8cx Gen 3") >= 0) {
```

Comment:
I don't understand this. If you are getting the SoC in the `SOC_EQ` by checking against `SC8280XP`, why checking this?

Reply:
The string used in the `SOC_EQ` array is `8cx Gen 3`. But Windows names the SOC `Snapdragon (TM) 8cx Gen 3` which is the only way to really identify the SoC on Windows. Nowhere can I seem to reliably find a string like `SC8280XP` or `8cx Gen 3` from Windows or any WinAPI functions that I know of.

I don't have any other WoA laptops to test on to determine if there is a pattern with how it generates `ProcessorNameString` but maybe on Windows in particular can trim after the `(TM)` so that `SOC_EQ` can match it?

It's really tricky here in particular, open to ideas.

Reply:
So if I understand you correctly, the `SOC_EQ` does nothing and the one with `strcmp` is the one actually matching.

If this is the case, please remove the `SOC_EQ` as it does nothing in your case (I want this PR to be Windows only). And checking against the "pretty" name that Windows give you is not okay in my opinion. In x86 what I do is print directly that value (which comes from CPUID, but in the end it's the same thing). In that case, the manufacturing process comes from the uarch but here it actually comes from the SoC, so that is an issue. 

If there is absolutely no way to get something that we can parse easily (e.g., `SC8280XP`) then I think the way to go is print the name from Windows directly and show Unknown under the manufacturing process. 

cpufetch is becoming increasingly similar to a database and that is something I don't want. And parsing the "pretty" name from Windows is going in that direction.

Reply:
One further comment about this.

I'd show just the SoC name coming from Windows and *hide* the manufacturing process instead of printing Unknown. You can add a condition for this in `src/common/printer.c` (just for Windows in ARM). I acknowledge the fact that it's not ideal not to get the manufacturing process but that way this will not look "worse" if we hide the Unknown.

At the end of the day, many tools just print the SoC name and not the manufacturing process!

Reply:
Just did some refactors. Found a way to at least try and determine the SoC vendor based on `ProcessorNameString`. The `soc_trademark_string` array can be used to try and derive the SoC vendor name with `strstr` in the new `try_match_soc_vendor_name` that I added.

![image](https://github.com/user-attachments/assets/1ed7d964-27ae-4443-872c-3eb7fdf8bab6)

Reply:
Smart, I like it, thanks :+1: